samples/src/main/java/com/example/bigquery/AddColumnLoadAppend.java

```diff
@@ -0,0 +1,102 @@
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobId;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.JobInfo.SchemaUpdateOption;
+import com.google.cloud.bigquery.JobInfo.WriteDisposition;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.LoadJobConfiguration;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+import com.google.common.collect.ImmutableList;
+import java.util.UUID;
+
+public class AddColumnLoadAppend {
+
+  public static void runAddColumnLoadAppend() {
+    // TODO(developer): Replace these variables before running the sample.
+    String datasetName = "MY_DATASET_NAME";
+    String tableName = "MY_TABLE_NAME";
+    String sourceUri = "/path/to/file.csv";
+    addColumnLoadAppend(datasetName, tableName, sourceUri);
+  }
+
+  public static void addColumnLoadAppend(String datasetName, String tableName, String sourceUri) {
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      TableId tableId = TableId.of(datasetName, tableName);
+      Table table = bigquery.getTable(tableId);
+
+      // Add a new column to a BigQuery table while appending rows via a load job.
+      // 'REQUIRED' fields cannot  be added to an existing schema, so the additional column must be
+      // 'NULLABLE'.
+      Schema newSchema =
+          Schema.of(
+              Field.newBuilder("name", LegacySQLTypeName.STRING)
+                  .setMode(Field.Mode.REQUIRED)
+                  .build(),
+              // Adding below additional column during the load job
+              Field.newBuilder("post_abbr", LegacySQLTypeName.STRING)
+                  .setMode(Field.Mode.NULLABLE)
+                  .build());
+
+      LoadJobConfiguration loadJobConfig =
+          LoadJobConfiguration.builder(tableId, sourceUri)
+              .setFormatOptions(FormatOptions.csv())
+              .setWriteDisposition(WriteDisposition.WRITE_APPEND)
+              .setSchema(newSchema)
+              .setSchemaUpdateOptions(ImmutableList.of(SchemaUpdateOption.ALLOW_FIELD_ADDITION))
+              .build();
+
+      // Create a job ID so that we can safely retry.
+      JobId jobId = JobId.of(UUID.randomUUID().toString());
+      Job loadJob = bigquery.create(JobInfo.newBuilder(loadJobConfig).setJobId(jobId).build());
+      System.out.println(loadJob.getJobId());
+
+      // Load data from a GCS parquet file into the table
+      // Blocks until this load table job completes its execution, either failing or succeeding.
+      Job completedJob = loadJob.waitFor();
+
+      // Check for errors
+      if (completedJob == null) {
+        throw new RuntimeException("Job no longer exists");
```

Comment:
nit: Probably InvalidStateException would make more sense?

RuntimeException is unchecked, making it a bad choice to suggest to users.

Reply:
InvalidStateException is not available so I will print out the error message instead

Reply:
I think there's a miscommunication on error handling here.

If something goes wrong, we definitely want to throw an exception - but we want a checked exception vs unchecked.

Check out this article for a breakdown of [checked v unchecked](https://howtodoinjava.com/java/exception-handling/checked-vs-unchecked-exceptions-in-java/).

Reply:
Throwing Exception so that it's checked.